ext/CatalystGraphMakieExtension/rn_graph_plot.jl

```diff
@@ -32,33 +36,46 @@ function SRGraphWrap(rn::ReactionSystem)
     end
     edgelist = vcat(collect(Graphs.edges(srg)), rateedges)
     edgeorder = sortperm(edgelist)
-    SRGraphWrap(srg, rateedges, edgeorder)
+    MultiGraphWrap(srg, rateedges, edgeorder)
+end
+
+# Automatically set edge order if not supplied
+function MultiGraphWrap(g::SimpleDiGraph{T}, multiedges::Vector{Graphs.SimpleEdge{T}}) where T
+    edgelist = vcat(collect(Graphs.edges(g)), multiedges)
+    edgeorder = sortperm(edgelist)
+    MultiGraphWrap(g, multiedges, edgeorder)
 end
 
-Base.eltype(g::SRGraphWrap) = eltype(g.g)
-Graphs.edgetype(g::SRGraphWrap) = edgetype(g.g)
-Graphs.has_edge(g::SRGraphWrap, s, d) = has_edge(g.g, s, d)
-Graphs.has_vertex(g::SRGraphWrap, i) = has_vertex(g.g, i)
-Graphs.inneighbors(g::SRGraphWrap{T}, i) where T = inneighbors(g.g, i)
-Graphs.outneighbors(g::SRGraphWrap{T}, i) where T = outneighbors(g.g, i)
-Graphs.ne(g::SRGraphWrap) = length(g.rateedges) + length(Graphs.edges(g.g))
-Graphs.nv(g::SRGraphWrap) = nv(g.g)
-Graphs.vertices(g::SRGraphWrap) = vertices(g.g)
-Graphs.is_directed(g::SRGraphWrap) = is_directed(g.g)
-
-function Graphs.edges(g::SRGraphWrap)
-    edgelist = vcat(collect(Graphs.edges(g.g)), g.rateedges)[g.edgeorder]
+Base.eltype(g::MultiGraphWrap) = eltype(g.g)
+Graphs.edgetype(g::MultiGraphWrap) = edgetype(g.g)
+Graphs.has_edge(g::MultiGraphWrap, s, d) = has_edge(g.g, s, d)
+Graphs.has_vertex(g::MultiGraphWrap, i) = has_vertex(g.g, i)
+Graphs.inneighbors(g::MultiGraphWrap{T}, i) where T = inneighbors(g.g, i)
+Graphs.outneighbors(g::MultiGraphWrap{T}, i) where T = outneighbors(g.g, i)
+Graphs.ne(g::MultiGraphWrap) = length(g.multiedges) + length(Graphs.edges(g.g))
+Graphs.nv(g::MultiGraphWrap) = nv(g.g)
+Graphs.vertices(g::MultiGraphWrap) = vertices(g.g)
+Graphs.is_directed(::Type{<:MultiGraphWrap}) = true
+Graphs.is_directed(g::MultiGraphWrap) = is_directed(g.g)
+
+function Graphs.edges(g::MultiGraphWrap)
+    edgelist = vcat(collect(Graphs.edges(g.g)), g.multiedges)[g.edgeorder]
 end
 
-function gen_distances(g::SRGraphWrap; inc = 0.2) 
+function gen_distances(g::MultiGraphWrap; inc = 0.25) 
     edgelist = edges(g)
     distances = zeros(length(edgelist))
-    for i in 2:Base.length(edgelist)
-        edgelist[i] == edgelist[i-1] && (distances[i] = inc)
+    for (i,e) in enumerate(edgelist)
+        fwd = findall(==(e), edgelist); rev = findall(==(reverse(e)), edgelist)
+        distances[i] != 0 && continue
+        distances[fwd] = collect(0:inc:inc*(length(fwd)-1))
+        distances[rev] = collect(inc:inc:inc*(length(rev)))
```

Comment:
Wrote a more efficient new way (that also correctly does the curves for multiple fwd/rev edges). One pass through the array and then one pass through the dict. Think it should be the same as storing the dict, since we construct a new MultiGraphWrap everytime we call `plot_network` and `plot_complexes` anyway